flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2767,6 +2770,11 @@ class ArrayExprLowering {
           auto lambda = genarr(*expr);
           operands.emplace_back([=](IterSpace iters) { return lambda(iters); });
         } break;
+        case Fortran::lower::LowerIntrinsicArgAs::Box: {
+          PushSemantics(ConstituentSemantics::RefOpaque);
+          auto lambda = genarr(*expr);
+          operands.emplace_back([=](IterSpace iters) { return lambda(iters); });
```

Comment:
I think you may want to do `return bldr->createBox(loc, lambda(iters);` here, otherwise I do not think the operand build for the elemental call will be a box.

Though, the intrinsics you are lowering does not allow testing the elemental aspect of `asBox`, so maybe simply leaving a TODO is safer. Otherwise, the asBox addition looks good to me.

Reply:
I've added the createBox call.